cunumeric/deferred.py

```diff
@@ -543,16 +544,9 @@ def _create_indexing_array(self, key, is_set=False):
             # the store with transformation
             rhs, store = self._copy_store(store)
 
-        if len(tuple_of_arrays) <= rhs.ndim and rhs.ndim > 1:
+        if len(tuple_of_arrays) <= rhs.ndim:
             output_arr = rhs._zip_indices(start_index, tuple_of_arrays)
             return True, rhs, output_arr, self
-        elif len(tuple_of_arrays) == 1 and rhs.ndim == 1:
-            key = tuple_of_arrays[0]
```

Comment:
Previously you would return a different type in the special case when `rhs.ndim == 1` (you would return an array of `coord_t`, instead of an array of `Point<1>`). But now we're always going through `_zip_indices`, so we now return `Point<1>`. Is the code after this OK with receiving this instead of `coord_t`?

Reply:
In case when` len(tuple_of_arrays) == 1 and rhs.ndim == 1` we use `key=tuple_of_arrays[0]` as an indirect region. Since it is 1d, internally its values converted to Point<1> when indirect copy is called (I think). Since we need to convert all negative indices for this case as well, I thought we should just call ZIP task for this case instead of implementing any special logic.

Reply:
I agree, I was just surprised this worked without a problem :-)